backend/src/routing/controllers/api/Facharbeit.ts

```diff
@@ -77,4 +82,43 @@ export default class Facharbeit {
             insertData(sql, args, defaultInsertCallback(res));
         };
     }
+    static POSTuploadDocument(req: Request, res: Response): void {
+        if (rejectWhenValidationsFail(req, res)) return;
+
+        const studentId = getStudentId();
+        const sql = 'SELECT id from facharbeiten WHERE studentId IS ?;';
+        // prüfen, ob der SuS eine Facharbeit angelegt hat
+        getFirstResult(sql, [studentId], (obj, err) => {
+            if (err) {
+                // Fehler bei der Abfrage (z.B. Datenbank nicht erreichbar)
+                res.status(500).json(err.message);
+                return;
+            } else if (obj) {
+                // SuS hat Facharbeit angelegt
+                // generate new storage name
+                const storageName = Date.now() + '.pdf';
+                const storagePath = path.join(FACHARBEIT_STORAGE, storageName);
```

Comment:
optimal wäre es, wenn du gleich die PDF in einen Ordner Speicherst mit Klasse Name Vorname des Schülers und dazu noch eine Textdatei mit abgabedatum in den Ordner legst. Siehe facharbeit flowchart. Würde das zip generieren leichter machen. 
Muss aber auch nicht sein

Reply:
Das passen wir bei Bedarf dann in https://trello.com/c/Nb4XbPYa/30-api-api-facharbeit-downloadall-get an. Zum Vorarbeiten verstehe ich das facharbeit flowchart nicht gut genug.